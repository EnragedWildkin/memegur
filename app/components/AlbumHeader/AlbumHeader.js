import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { Header, Icon, Right, Left, Button } from 'native-base';
import { Actions } from 'react-native-router-flux';
import styles from './styles';

export default class AlbumHeader extends Component {
  render() {
    const iconStyle = StyleSheet.flatten(styles.iconStyle);
    const topNavBarStyle = StyleSheet.flatten(styles.topNavBar);

    return (
      <Header style={topNavBarStyle}>
        <Left>
          <Button transparent onPress={Actions.Gallery}>
            <Icon name="arrow-back" style={iconStyle} />
          </Button>
        </Left>

        <Right>
          <Icon name="md-arrow-round-up" style={iconStyle} />
          <Icon name="md-arrow-round-down" style={iconStyle} />
          <Icon name="heart" style={iconStyle} />
          <Icon name="md-share" style={iconStyle} />
        </Right>
      </Header>
    );
  }
}
