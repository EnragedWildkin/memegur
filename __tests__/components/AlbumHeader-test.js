import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import AlbumHeader from '../../app/components/AlbumHeader/AlbumHeader';

describe('<AlbumHeader />', () => {
  it('renders post header', () => {
    const tree = renderer.create(
      <AlbumHeader />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
