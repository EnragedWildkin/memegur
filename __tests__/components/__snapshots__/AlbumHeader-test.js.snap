// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumHeader /> renders post header 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#33363f",
        "borderBottomColor": "#33363f",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 15,
        "right": 0,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "top": 0,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "alignSelf": "center",
          "flex": 1,
        }
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginLeft": -3,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "white",
                "fontSize": 33,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 2,
                "padding": 5,
                "paddingTop": 2,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          virtual={undefined}
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "white",
              "fontSize": 30,
              "padding": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "white",
              "fontSize": 30,
              "padding": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "white",
              "fontSize": 30,
              "padding": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "white",
              "fontSize": 30,
              "padding": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
    </View>
  </View>
</View>
`;
